gcp/main.tf

```diff
@@ -0,0 +1,257 @@
+    ports    = ["0-65535"]
+  }
+  allow {
+    protocol = "udp"
+    ports    = ["0-65535"]
+  }
+
+  source_ranges = ["${var.vpc_network_cidr}"]
+}
+resource "google_compute_firewall" "external" {
+  name    = "${google_compute_network.vpc_network.name}-external"
+  network = "${google_compute_network.vpc_network.name}"
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22", "8080", "9200"]
+  }
+
+  target_tags   = ["humios", "kafkas"]
+  source_ranges = "${var.external_access_ips}"
+}
+resource "google_compute_firewall" "allow_health_check" {
+  name    = "${google_compute_network.vpc_network.name}-health-check"
+  network = "${google_compute_network.vpc_network.name}"
+
+  allow {
+    protocol = "tcp"
+    ports    = ["8080", "9200"]
+  }
+
+  target_tags   = ["humios", "kafkas"]
+  source_ranges = ["130.211.0.0/22", "35.191.0.0/16"]
+}
+
+
+resource "google_compute_global_address" "default" {
+  name = "humio-global-ip"
+}
+resource "google_compute_global_address" "es" {
+  name = "humio-global-es-ip"
+}
+resource "google_compute_global_forwarding_rule" "https" {
+  name       = "humio-global-forward"
+  target     = "${google_compute_target_https_proxy.default.self_link}"
+  ip_address = "${google_compute_global_address.default.address}"
+  port_range = "443"
+}
+resource "google_compute_global_forwarding_rule" "es" {
+  name       = "humio-global-forward-es"
+  target     = "${google_compute_target_https_proxy.es.self_link}"
+  ip_address = "${google_compute_global_address.es.address}"
+  port_range = "443"
+}
+
+
+resource "google_compute_http_health_check" "default" {
+  name               = "default"
+  request_path       = "/"
+  check_interval_sec = 1
+  timeout_sec        = 1
+  port = 8080
+}
+resource "google_compute_health_check" "es" {
+  name               = "escheck"
+  check_interval_sec = 1
+  timeout_sec        = 1
+  tcp_health_check {
+    port = "9200"
+  }
+}
+
+
+// create an pd-ssd for each humio-host
+resource "google_compute_disk" "humio-pd-ssd-" {
+    count   = "${var.instances}"
+    name    = "humio-pd-ssd-${count.index}-data"
+    type    = "pd-ssd"
+    zone    = "${var.zone}"
+    size    = "${var.humio_disk_size}"
+}
+
+resource "google_compute_instance" "humios" {
+ count = "${var.instances}"
+ name = "${format("humio%02d-%s", count.index + 1, var.region)}"
+ machine_type = "${var.machine_type}" 
+ zone         = "${var.zone}"
+
+ attached_disk {
+    source      = "${element(google_compute_disk.humio-pd-ssd-.*.self_link, count.index)}"
+    device_name = "${element(google_compute_disk.humio-pd-ssd-.*.name, count.index)}"
+ }
+
+ boot_disk {
+   initialize_params {
+     image = "${var.boot_disk_image}" 
+     size = "${var.boot_disk_size}"
+   } 
+ }
+
+ metadata {
+   sshKeys = "${var.ansible_ssh_user}:${var.access_pub_key}"
+ }
+
+ tags = [
+        "${count.index < var.zookeepers ? "zookeepers" : "no-zookeepers"}",
+        "kafkas",
+        "humios"
+  ]
+
+
+ metadata_startup_script = <<SCRIPT
```

Comment:
Personally I'm starting to prefer cloudinit which in theory, at least, should make it easier to switch distribution.